### ISSUE TITLE
Custom headers are not passed to WebSocket connection request

### ISSUE THREAD

**Issue by maximbaz** [qutebrowser/qutebrowser#6402]
**Version info**:
```
qutebrowser v2.2.0
Git commit: fe80b04ed on master (2021-04-17 10:43:52 +0200)
Backend: QtWebEngine 5.15.3, Chromium 87.0.4280.144
Qt: 5.15.2
```

**Does the bug happen if you start with `--temp-basedir`?**: Yes

**Description**

`content.headers.custom` are not being passed to the request that initiates websocket connection.

**How to reproduce**

1. `python3 -m qutebrowser --temp-basedir`
1. `:set content.headers.custom '{"hello": "world"}'`
1. Open devtools and then open `https://www.websocket.org/echo.html`
1.  See that main request, js, images, etc, they all have "hello=world" in the request header
1. Clear network tab (for simplicity), and click on "Connect" button

I expected to see "hello=world" in the request headers of the new request, but it is not there.

**Comment by The-Compiler** [comment]
Reported upstream: [[QTBUG-92932] [REG 5.12 -> 5.13] QWebEngineUrlRequestInterceptor doesn't get called for websocket connections - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-92932)

**Comment by maximbaz** [comment]
Thanks @The-Compiler! There is a related issue which I accidentally stumbled upon while looking for repro steps above, I have a feeling it's also upstream bug but maybe you could double-check and forward if that's the case?

1. Run first instance: `python3 -m qutebrowser --temp-basedir`, open devtools and then go to `linkedin.com`
2. Run second instance: `python3 -m qutebrowser --temp-basedir -s content.headers.custom '{"hello": "world"}'`, open devtools and go to `linkedin.com`

Compare network tab of both devtools - the second instance will have a lot of CORS-related errors, while the first instance loads everything correctly.

**Comment by The-Compiler** [comment]
Thanks, I opened #6412 - will have a look, but busy with other stuff right now.

**Comment by The-Compiler** [comment]
Looks like a fix will follow in Qt 6.2 (`dev` branch): [WIP: Apply url request interceptor on websocket connect as well (I01bc8566) · Gerrit Code Review](https://codereview.qt-project.org/c/qt/qtwebengine/+/343977)
